torch/fx/experimental/symbolic_shapes.py

```diff
@@ -217,28 +217,42 @@ def sym_and(self, other):
     def int_(self):
         if len(self.expr.free_symbols) == 0:
             return int(self.expr)
-        raise RuntimeError("Trying to extract a concrete int out of a symbolic int")
+        raise RuntimeError(f"Trying to extract a concrete int out of a symbolic int {self.expr}")
 
     # You can manually trigger a guard with this function
     def guard_int(self, file, line):
         # TODO: use the file/line for some useful diagnostic on why a
         # guard occurred
-        return int(self.shape_env.evaluate_expr(self.expr))
+        r = self.shape_env.evaluate_expr(self.expr)
+        try:
+            return int(r)
+        except Exception:
```

Comment:
Why catch BareExceptions instead of just RuntimeErrors?

Reply:
Because I was lazy and didn't look up what the actual class of sympy errors could be raised